### PR TITLE
SFINT-4347 Headless logic inside SubjectInput and DescriptionInput

### PR DESCRIPTION
**Pull request by mmitiche**
## Headless logic integrated with `SubjecInput `and `DescriptionInput` components

https://user-images.githubusercontent.com/86681870/150133041-051d3cf8-2cd7-4a6b-8a86-7d964c149c9a.mov

### CODE REVIEW

File: src/main/default/lwc/descriptionInput/descriptionInput.js [hunk 1]
```diff
@@ -65,28 +129,73 @@ export default class DescriptionInput extends LightningElement {
...
-   * Shows an error message in the component if there is an error.
-   * @returns {void}
+   * Returns rich text value after replacing the HTML tags with spaces.
+   * @param {string} htmlStr
+   * @returns {string}
    */
-  @api validate() {
-    this._validity = !this.required || !!this._value;
+  replaceTagsWithSpace(htmlStr) {
+    return htmlStr.replace(/(<([^>]+)>)/gi, ' ');
```
Comment (lbergeron) [inline src/main/default/lwc/descriptionInput/descriptionInput.js:199]: Is the inner capture group necessary, it doesn't seem to be?

Also, the regex won't replace this string `<>`.

```suggestion
    return htmlStr.replace(/(<[^>]*>)/gi, ' ');
```

**Review by lbergeron — APPROVED** [review]
Added a few comments, but nothing major. It looks good!

File: src/main/default/lwc/descriptionInput/descriptionInput.js [hunk 2]
```diff
@@ -56,6 +71,55 @@ export default class DescriptionInput extends LightningElement {
     'table',
     'header'
   ];
+  /**
+   * The necessary length that the description must have in order to be considered strong.
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply src/main/default/lwc/descriptionInput/descriptionInput.js]: Updated to measure strength in words as suggested 👌

File: src/main/default/lwc/descriptionInput/descriptionInput.js [hunk 3]
```diff
@@ -1,45 +1,60 @@
...
+import { debounce } from 'c/utils';
+
+/** @typedef {import("coveo").CaseAssistEngine} CaseAssistEngine */
+/** @typedef {import("coveo").CaseInput} CaseInput */
 
 /**
  * The `descriptionInput` component displays a rich text input for the case description.
  * @example
- * <c-description-input label="Explain the problem" messageWhenValueMissing="Complete this field." required></c-description-input>
+ * <c-description-input engineId={engineId} caseEditDelayMs="500" label="Explain the problem" messageWhenValueMissing="Complete this field." required displayStrengthIndicator></c-description-input>
```
Comment (erocheleau) [inline src/main/default/lwc/descriptionInput/descriptionInput.js:17]: BTW @mmitiche all the attributes, defined with `lowerUpper` in your component map to `lower-upper` in the HTML when you use it.

Comment (erocheleau) [reply src/main/default/lwc/descriptionInput/descriptionInput.js:17]:
```suggestion
 * <c-description-input engine-id={engineId} case-edit-delay-ms="500" label="Explain the problem" message-when-value-missing="Complete this field." required display-strength-indicator></c-description-input>
```

File: src/main/default/lwc/descriptionInput/__tests__/descriptionInput.test.js [hunk 4]
```diff
@@ -74,6 +74,27 @@ describe('c-description-input', () => {
...
+    const element = createTestComponent();
+
+    await allPromisesResolution();
+    const strengthIndicatorNode = element.shadowRoot.querySelector(
+      'c-description-strength-indicator'
+    );
+    expect(strengthIndicatorNode).toBeNull();
+  });
+
+  it('should display the description strength indicator when #displayStrengthIndicator attribute is specified', async () => {
```
Comment (erocheleau) [inline src/main/default/lwc/descriptionInput/__tests__/descriptionInput.test.js:87]: why is there a # ?

**Review by erocheleau — APPROVED** [review]

**Pull request merged**